### PR TITLE
merge r1.11 to main

### PR DESCRIPTION
**Pull request by XuesongYang**
# What does this PR do ?

This PR is to merge `r1.11.0` branch to the `main` branch. There were merge conflicts according to several updates in all domains. I fixed all conflicts. For some conflicts, I chose the `main` changes, while for other conflicts, I made a decision after reading detailed changes in both `r1.11.0` and `main`. 

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# Before your PR is "Ready for review"
**Pre checks**:
- [] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [] Did you write any new necessary tests?
- [] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?
Owners of PRs in every domain, please help to review the changes. Thanks.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: tutorials/tts/FastPitch_MixerTTS_Training.ipynb [hunk 1]
```diff
@@ -1,612 +1,611 @@
...
-  {
-   "cell_type": "code",
-   "execution_count": null,
-   "id": "7ef87e31",
-   "metadata": {},
-   "outputs": [],
-   "source": [
-    "from nemo.collections.tts.torch.data import TTSDataset\n",
-    "from nemo_text_processing.text_normalization.normalize import Normalizer\n",
-    "from nemo_text_processing.g2p.modules import EnglishG2p\n",
```
Comment (redoctopus) [inline tutorials/tts/FastPitch_MixerTTS_Training.ipynb:201]: I believe `main` has the correct path here (i.e. `nemo_text_processing.g2p.modules`) and is more up-to-date.

**Review by redoctopus — CHANGES_REQUESTED** [review]
I believe there are a few files that need to be kept from `main` due to the change in G2P module locations from #4690, @AlexGrinch could you take a look to check what files need to be updated?

A few that I've caught that should be kept as the version from `main`:
- `nemo/collections/tts/torch/g2ps.py`
- `nemo/collections/tts/torch/tts_dataset.yaml`
- `nemo/collections/tts/models/fastpitch.py`
- `nemo/collections/tts/torch/tts_tokenizers.py`
- `tutorials/tts/FastPitch_MixerTTS_Training.ipynb`

File: tutorials/tts/FastPitch_MixerTTS_Training.ipynb (hunk 1 above)
Comment (XuesongYang) [reply tutorials/tts/FastPitch_MixerTTS_Training.ipynb:201]: Thanks for the info! i will fix the rules for yaml and tutorial, that keep main in main branch.

**Review by ericharper — CHANGES_REQUESTED** [review]
Blocking this as I'm seeing a bunch of CI tests are being skipped but the CI is showing as passed.

File: nemo/package_info.py [hunk 2]
```diff
@@ -16,7 +16,7 @@
 MAJOR = 1
 MINOR = 12
 PATCH = 0
-PRE_RELEASE = 'rc0'
+PRE_RELEASE = ''
```
Comment (ericharper) [inline nemo/package_info.py:19]: This should be 'rc0'

**Review by ericharper — DISMISSED** [review]
LGTM. Thanks!

File: tutorials/tts/FastPitch_MixerTTS_Training.ipynb (hunk 1 above)
Comment (XuesongYang) [reply tutorials/tts/FastPitch_MixerTTS_Training.ipynb:201]: fixed.

File: tutorials/tts/FastPitch_MixerTTS_Training.ipynb [hunk 3]
```diff
@@ -183,32 +165,23 @@
...
    "source": [
+    "from nemo_text_processing.g2p.modules import EnglishG2p\n",
     "from nemo.collections.tts.torch.data import TTSDataset\n",
     "from nemo_text_processing.text_normalization.normalize import Normalizer\n",
-    "from nemo_text_processing.g2p.modules import EnglishG2p\n",
-    "from nemo.collections.common.tokenizers.text_to_speech.tts_tokenizers import (\n",
-    "    EnglishPhonemesTokenizer,\n",
-    "    EnglishCharsTokenizer,\n",
-    ")"
+    "from nemo.collections.tts.torch.tts_tokenizers import EnglishPhonemesTokenizer, EnglishCharsTokenizer"
```
Comment (redoctopus) [inline tutorials/tts/FastPitch_MixerTTS_Training.ipynb:180]: I still see the import here as coming from the TTS collection, could you revert this back to the path in `main`?

Comment (XuesongYang) [reply tutorials/tts/FastPitch_MixerTTS_Training.ipynb]: yes. Fixed already, forgot to push.

**Review by vadam5 — APPROVED** [review]
LGTM

**Comment by XuesongYang** [comment]
> Blocking this as I'm seeing a bunch of CI tests are being skipped but the CI is showing as passed.

Thanks for pointing it out. I rebased to the latest main. It would be related to the most recent PR that removed computer vision codes and tests.
https://github.com/NVIDIA/NeMo/pull/4907/files

File: README.rst [hunk 4]
```diff
@@ -200,16 +200,10 @@ Megatron GPT training requires NVIDIA Apex to be installed.
 
 .. code-block:: bash
 
-    git clone https://github.com/NVIDIA/apex
+    git clone https://github.com/ericharper/apex.git
```
Comment (XuesongYang) [inline README.rst:203]: @ericharper do we need to take care of this change? this apex seems hacked a little bit from its main.

Comment (ericharper) [reply README.rst:203]: No this was done to make installation easier.

**Pull request merged**